lib/jekyll/admin/pages/handler.rb

```diff
@@ -0,0 +1,46 @@
+module Jekyll
+  module Admin
+    module Pages
+      class Handler < ApiHandler
+        def index
+          pages = []
+          @site.pages.each do |page|
+            next if page.data["title"].nil?
+            extname = File.extname(page.name)
+            page_id = File.basename(page.name, extname)
+            meta = parse_frontmatter(page.path)
+            page_item = {
```

Comment:
Whenever possible, I'd call `page.to_liquid.to_json` (to use Jekyll's native representation), even if we're then only taking a subset of that for e.g., indexes. That way our internal data models will be future proof to Jekyll's upstream changes.